diff --git a/Makefile.am b/Makefile.am
index e5d0a4d..e5bfa67 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -69,8 +69,8 @@ mod_auth_openidc.la: libauth_openidc.la
 	${APXS} -c -o $@ $< ${AM_CFLAGS} ${LIBADD}
 
 install-exec-local:
-	@APXS@ -i -a -n auth_openidc mod_auth_openidc.la
-#	${INSTALL} -p -m 755 -D .libs/mod_auth_openidc.so @APACHE_MODULEDIR@/mod_auth_openidc.so
+#	@APXS@ -i -a -n auth_openidc mod_auth_openidc.la
+	${INSTALL} -p -m 755 -D .libs/mod_auth_openidc.so $(DESTDIR)@APACHE_MODULEDIR@/mod_auth_openidc.so
 
 LDADD = libauth_openidc.la ${LIBADD}
 
